tap_bitbucket/__init__.py

```diff
@@ -0,0 +1,891 @@
+                pr['source'] = normalize_pull_request_endpoint(pr['source'])
+                pr['destination'] = normalize_pull_request_endpoint(pr['destination'])
+
+                with singer.Transformer() as transformer:
+                    rec = transformer.transform(pr, schemas['pull_requests'], metadata=metadata.to_map(mdata))
+                singer.write_record('pull_requests', rec, time_extracted=extraction_time)
+                counter.increment()
+
+                if schemas.get('pull_request_comments'):
+                    sync_all_pull_request_comments(schemas, org, repo_path, pr['id'], pr['number'], state, mdata, start_date)
+
+    singer.write_bookmark(state, repo_path, 'pull_requests', {
+        'since': singer.utils.strftime(extraction_time)
+    })
+
+    return state
+
+def sync_all_pull_request_comments(schemas, org, repo_path, pr_id, pr_number, state, mdata, start_date):
+    # bookmark data is keyed by pr ID, and the value is the date/time of the most recent successful ingest
+    bookmark = get_bookmark(state, repo_path, "pull_request_comments", pr_id, start_date)
+    if not bookmark:
+        bookmark = '1970-01-01'
+    bookmarkTime = parser.parse(bookmark)
+    if bookmarkTime.tzinfo is None:
+        bookmarkTime = pytz.UTC.localize(bookmarkTime)
+
+    with metrics.record_counter('pull_request_comments') as counter:
+        extraction_time = singer.utils.now()
+        query = urllib.parse.quote('created_on>{}'.format(bookmarkTime.isoformat()))
+        for response in authed_get_all_pages(
+            'pull_request_comments',
+            'https://api.bitbucket.org/2.0/repositories/{}/pullrequests/{}/comments?'.format(repo_path, pr_number) + \
+                'q={}'.format(query)
+        ):
+            comments = response.json()['values']
+            for comment in comments:
+                comment['_sdc_repository'] = repo_path
+                comment['id'] = '{}/{}'.format(pr_id, comment['id'])
+                comment['pr_id'] = pr_id
+                
+                parent = comment.get('parent')
+                if parent:
+                    parent['id'] = '{}/{}'.format(pr_id, parent['id'])
+
+                with singer.Transformer() as transformer:
+                    rec = transformer.transform(comment, schemas['pull_request_comments'], metadata=metadata.to_map(mdata))
+                singer.write_record('pull_request_comments', rec, time_extracted=extraction_time)
+                counter.increment()
+
+    singer.write_bookmark(state, repo_path, 'pull_request_comments', {
+        pr_id: singer.utils.strftime(extraction_time)
+    })
+
+    return state
+
+def sync_all_repositories(schema, repo_path, state, mdata, _start_date):
+    repo_metadata = get_repo_metadata(repo_path)
+
+    with metrics.record_counter('repositories') as counter:
+        extraction_time = singer.utils.now()
+        repo = {}
+        repo['id'] = 'bitbucket:' + repo_path
```

Comment:
We are using a slash for other sources instead of a colon, so it'd probably be best to keep that consistent.

Reply:
Long story short, the `repositories` stream is a special-purpose ingest that dumps into [target-reposync](https://github.com/minwareco/target-reposync). That target is expecting the `:` delimiter, based on the way I coded it at the time.

I'm fine with using "/" for consistency, so have updated `target-reposync` to use that (and it still supports ":" for backward compat with the existing taps until we may update them). The bitbucket tap here has been updated to use "/" accordingly.